Source/WPEFramework/PluginServer.cpp

```diff
@@ -290,8 +290,21 @@ namespace PluginHost {
             AddRef();
             result = static_cast<PluginHost::IShell*>(this);
         }
-        else {
+        else if (id == PluginHost::IShell::ICOMLink::ID) {
+            AddRef();
+            result = static_cast<PluginHost::IShell::ICOMLink*>(this);
+        }
+        else if (id == PluginHost::IShell::IConnectionServer::ID) {
+            _pluginHandling.Lock();
 
+            if (_jsonrpc != nullptr) {
```

Comment:
Do we want to limit this to only available if there is JSONRPC on the Plugin? It is now generic functionality...

Reply:
Totally agree. Additionally you may still be interested in connections, but not necessarily implement provide JSONRPC yourself.